v2/stream/new/options.go

```diff
@@ -0,0 +1,147 @@
+package new
+
+import (
+	"os"
+	"time"
+
+	"github.com/alpacahq/alpaca-trade-api-go/common"
+)
+
+type options struct {
+	logger         Logger
+	host           string
+	key            string
+	secret         string
+	reconnectLimit int
+	reconnectDelay time.Duration
+	processorCount int
+	bufferSize     int
+	trades         []string
+	tradeHandler   func(Trade)
+	quotes         []string
+	quoteHandler   func(Quote)
+	bars           []string
+	barHandler     func(Bar)
+}
+
+func (o *options) apply(opts ...Option) {
+	for _, opt := range opts {
+		opt.apply(o)
+	}
+}
+
+type Option interface {
+	apply(*options)
+}
+
+type funcOption struct {
+	f func(*options)
+}
+
+func (fo *funcOption) apply(o *options) {
+	fo.f(o)
+}
+
+func newFuncOption(f func(*options)) *funcOption {
+	return &funcOption{
+		f: f,
+	}
+}
+
+// defaultOptions are the default options for a client.
+// Don't change this in a backward incompatible way!
+func defaultOptions() *options {
+	host := "https://stream.data.alpaca.markets"
+	if s := os.Getenv("DATA_PROXY_WS"); s != "" {
+		host = s
+	}
+
+	return &options{
+		logger:         newStdLog(),
+		host:           host,
+		key:            common.Credentials().ID,
+		secret:         common.Credentials().Secret,
+		reconnectLimit: 20,
+		reconnectDelay: 150 * time.Millisecond,
+		processorCount: 1,
+		bufferSize:     100000,
+		trades:         []string{},
+		tradeHandler:   func(t Trade) {},
+		quotes:         []string{},
+		quoteHandler:   func(q Quote) {},
+		bars:           []string{},
+		barHandler:     func(b Bar) {},
+	}
+}
+
+// WithLogger configures the logger
+func WithLogger(logger Logger) Option {
+	return newFuncOption(func(o *options) {
+		o.logger = logger
+	})
+}
+
+// WithHost configures the host
+func WithHost(host string) Option {
+	return newFuncOption(func(o *options) {
+		o.host = host
+	})
+}
+
+// WithCredentials configures the key and secret to use
+func WithCredentials(key, secret string) Option {
+	return newFuncOption(func(o *options) {
+		o.key = key
+		o.secret = secret
+	})
+}
+
+// WithReconnectSettings configures how many consecutive connection
+// errors should be accepted and the delay between retries
```

Comment:
Please add that the delay is actually multiplied by the consecutive error count!